### PR TITLE
CephFS: Add clusterName to snapshot metadata and honor `--setmetadata` option

### PR DESCRIPTION
**Pull request by pkalever**
# Describe what this PR does #

* Add clusterName as metadata on snapshots

Example:
```
sh-4.4$ ceph fs subvolume snapshot metadata ls myfs csi-vol-ba248f9e-0e75-11ed-b774-8e97192ff5ec \
                        csi-snap-ce24e3bb-0e75-11ed-b774-8e97192ff5ec --group_name csi
{
    "csi.ceph.com/cluster/name": "K8s-cluster-1",
    "csi.storage.k8s.io/volumesnapshot/name": "cephfs-pvc-snapshot",
    "csi.storage.k8s.io/volumesnapshot/namespace": "rook-ceph",
    "csi.storage.k8s.io/volumesnapshotcontent/name": "snapcontent-2e89e1b2-e6e9-48fe-b365-edb493d7022e"
}
```

* honor `--setmetadata` option at CephFS metadata code
`--setmetadata` is false by default, honoring it will keep the metadata disabled by default

Signed-off-by: Prasanna Kumar Kalever \<prasanna.kalever@redhat.com\>

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)
- `/retest all`: run this in case the CentOS CI failed to start/report any test
  progress or results

</details>

### CODE REVIEW

File: internal/cephfs/controllerserver.go [hunk 1]
```diff
@@ -151,6 +154,7 @@ func checkContentSource(
 	req *csi.CreateVolumeRequest,
 	cr *util.Credentials,
 	clusterName string,
+	setMetadata bool,
```
Comment (nixpanic) [inline internal/cephfs/controllerserver.go]: you can make this `checkContentSource()` function a method of `ControllerServer`. That way there is no need to pass the new two arguments `clusterName` and `setMetadata`. They can then be used directly as `cs.ClusterName` and `cs.SetMetadata`.

File: internal/cephfs/controllerserver.go [hunk 2]
```diff
@@ -538,12 +543,13 @@ func cleanUpBackingVolume(
 	volID *store.VolumeIdentifier,
 	cr *util.Credentials,
 	clusterName string,
+	setMetadata bool,
```
Comment (nixpanic) [inline internal/cephfs/controllerserver.go]: same here, consider making `cleanUpBackingVolume()` a method of `ControllerServer` to reduce the number of arguments that the function requires

File: internal/cephfs/controllerserver.go [hunk 3]
```diff
@@ -896,12 +888,13 @@ func doSnapshot(
 	volOpt *store.VolumeOptions,
 	snapshotName,
 	clusterName string,
+	setMetadata bool,
```
Comment (nixpanic) [inline internal/cephfs/controllerserver.go]: lots of parameters here too, consider making it a method of `ControllerServer`

**Review by nixpanic — COMMENTED** [review]
Added some notes to clean things up a little, and keep it a little easier to maintain in the future

**Comment by pkalever** [comment]
@nixpanic addressed the review comments, PTAL. Thanks!

**Review by nixpanic — APPROVED** [review]

**Comment by pkalever** [comment]
@Mergifyio rebase

**Comment by pkalever** [comment]
@Mergifyio rebase

**Review by Madhu-1 — APPROVED** [review]

**Review by yati1998 — APPROVED** [review]

**Review by Rakshith-R — APPROVED** [review]

**Pull request merged**